parsers/GB.py

```diff
@@ -1,71 +1,99 @@
-#!/usr/bin/env python3
-
-"""
-Parser that uses the RTE-FRANCE API to return the following data type(s)
-fetch_price method copied from FR parser.
-Day-ahead Price
-"""
+"""Parser that uses the RTE-FRANCE API"""
 
 import xml.etree.ElementTree as ET
-from datetime import datetime, timedelta
+from datetime import datetime, timedelta, timezone
 from logging import Logger, getLogger
+from zoneinfo import ZoneInfo
 
-import arrow
 from requests import Session
 
+from electricitymap.contrib.lib.models.event_lists import PriceList
+from electricitymap.contrib.lib.models.events import EventSourceType
+from electricitymap.contrib.lib.types import ZoneKey
 from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+PARSER = "GB.py"
+TIMEZONE = ZoneInfo("Europe/London")
+ZONE_KEY = ZoneKey("GB")
 
 
 @refetch_frequency(timedelta(days=1))
 def fetch_price(
-    zone_key: str,
+    zone_key: ZoneKey = ZONE_KEY,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list:
-    if target_datetime:
-        now = arrow.get(target_datetime, tz="Europe/Paris")
-    else:
-        now = arrow.now(tz="Europe/London")
+) -> list[dict]:
+    """Returns one-hourly prices for the requested day."""
 
-    r = session or Session()
-    formatted_from = now.shift(days=-1).format("DD/MM/YYYY")
-    formatted_to = now.format("DD/MM/YYYY")
+    now = datetime.now(timezone.utc)
+    target_datetime = (
+        now if target_datetime is None else target_datetime.astimezone(timezone.utc)
+    )
```

Comment:
thought: The previous code was returning two days of data, which I thought was a bug so I changed it to just one. 

In light of the recent comments in the other PRs though it was probably a feature 😅 Should I revert? (and would I need to bump the refetch frequency to `2` then?)

Reply:
It would be nice with 2 days of data. But it's not critical.

Reply:
Will do and update the snapshots - shouldn't be too much of a problem